pkg/apis/smith/v1/types.go

```diff
@@ -194,11 +197,27 @@ type Resource struct {
 	Name ResourceName `json:"name"`
 
 	// Explicit dependencies.
-	DependsOn []ResourceName `json:"dependsOn,omitempty"`
+	References []Reference `json:"reference,omitempty"`
```

Comment:
I prefer plural `references` in JSON.
Also, we already have plural `resources` in the BundleSpec:
```go
type BundleSpec struct {
	Resources []Resource `json:"resources,omitempty"`
}
```